src/main/scala/redis/clients/jedis/Implicits.scala

```diff
@@ -0,0 +1,24 @@
+package redis.clients.jedis
+
+import java.util.{Map => JMap}
+
+/**
+  * @author The Viet Nguyen
+  */
+object Implicits {
+
+  implicit class KvPipeline(delegate: Pipeline) {
```

Comment:
I don't see the usage of this except tests. Is it some preparation for the 'Don't include primary key to redis hash' story?

Reply:
yes

Reply:
The pipeline does not store information about the key, so we need to tweak it a little bit. I can create a separate pull request if you would like to

Reply:
Just a caution: If we run 10K operations with getHashAllWithKey() with a pipeline it will create 10K instances of KvPipeline class, so we will need to create KvPipeline once rather than using implicits.

Reply:
Yep, could you please create a separate PR for this functionality? Ideally together with the changes where you use this utility class.